src/security/builtin_plugins/cryptographic/src/crypto_transform.c

```diff
@@ -555,14 +579,14 @@ static bool read_secure_rtps_body(unsigned char **ptr, unsigned char *endp, DDS_
   {
     body->data.base = *ptr;
 
-    if (!read_subnsg_header(ptr, endp, SMID_INFO_SRC, &smhdr, &bswap))
+    if (!read_submsg_header(ptr, endp, SMID_INFO_SRC, &smhdr, &bswap))
       return false;
 
     *ptr += smhdr.octetsToNextHeader;
 
-    while ((uintptr_t)*ptr + sizeof(smhdr) < (uintptr_t)endp)
+    while (sizeof(smhdr) < (size_t)(endp - *ptr))
```

Comment:
Shouldn't this be ≤?

I also wonder which submessage kinds can be encountered here, and whether there `octetsToNextHeader` fields are always straightforward. In particular, I'm thinking of `PAD` and `SENTINEL`, and about the special case of `octetsToNextHeader` being 0 (for submessages other than `PAD`).